### PR TITLE
Hydrate with the Python Project Template

### PR DESCRIPTION
**Pull request by camposandro**
Pulls the LINCC Frameworks's [Python Project Template](https://github.com/lincc-frameworks/python-project-template) to enhance software engineering practices.

## Added features

- Unit tests with coverage (and deployment to codecov)
- Deployment of code documentation to Read The Docs (with support to render notebooks)
- Publishing package with dynamic versioning (using GitHub release tags and PyPI's Trusted Publishers)
- Pre-commit hooks to increase compliance during development

### TODOs after merging:

- Connect codecov and Read The Docs to the repository
- Create trusted publisher in gundam PyPI project
- Update README badges
- Fix issues reported by ruff in the pre-commit hooks

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,29 +6,6 @@ __pycache__/
...
 *.so
 
-# Specific intermediate files created by f2pỳ
-cflibformodule.c
-cflibfor-f2pywrappers2.f90
-
-# Benchs
-bench/
-
-# Files output by Gundam by default
```
Comment (delucchi-cmu) [inline .gitignore:16]: Let's keep these in the ignore file, too.

File: README.md [hunk 2]
```diff
@@ -1,4 +1,12 @@
-# <img src="./docs/banner.png">
+# <img src="./docs/images/banner.png">
+
+[![Template](<https://img.shields.io/badge/Template-LINCC%20Frameworks%20Python%20Project%20Template-brightgreen>)](<https://lincc-ppt.readthedocs.io/en/latest/>)
+[![PyPI](<https://img.shields.io/pypi/v/gundam?color=blue&logo=pypi&logoColor=white>)](<https://pypi.org/project/gundam/>)
+[![Read the Docs](<https://img.shields.io/readthedocs/gundam>)](<https://gundam.readthedocs.io/>)
+[![Actions Status](https://github.com/camposandro/gundam/actions/workflows/testing-and-coverage.yml/badge.svg)](<https://gundam.readthedocs.io/>)
```
Comment (delucchi-cmu) [inline README.md]: "Actions", "Documentation" and "Linter" actions should link to their respective action status. I guess there's a chicken-and-egg problem here, though...

**Review by delucchi-cmu — COMMENTED** [review]
Woah! This is awesome!

File: .gitignore (hunk 1 above)
Comment (camposandro) [reply .gitignore:16]: Thanks, I added them back!

File: README.md (hunk 2 above)
Comment (camposandro) [reply README.md]: I also think the amount of badges here was also unnecessary. I cleaned these up and fixed the links to point to @samotracio's repository!

**Pull request merged**